src/main/java/sirius/biz/sequences/Sequences.java

```diff
@@ -0,0 +1,176 @@
+            }
+
+            throw Exceptions.handle()
+                            .to(LOG)
+                            .withSystemErrorMessage(
+                                    "Failed to generate an unique sequence number for %s. Giving up after 25 retries.",
+                                    sequence)
+                            .handle();
+        } catch (SQLException e) {
+            throw Exceptions.handle()
+                            .to(LOG)
+                            .error(e)
+                            .withSystemErrorMessage(
+                                    "Failed to generate an unique number for %s due to a database error: %s",
+                                    sequence)
+                            .handle();
+        }
+    }
+
+    /**
+     * Sets the initial or next value of the given sequence.
+     * <p>
+     * Unless <tt>force</tt> is set to <tt>true</tt>, the value has to be higher than the current counter value to
+     * prevent non unique numbers from being generated.
+     *
+     * @param sequence  the sequence to update
+     * @param nextValue the next  value that will be returned when calling {@link #generateId(String)} for this
+     *                  sequence.
+     * @param force     if <tt>true</tt>, no sanity checks are performed and the sequence can be reset to <b>ANY</b>
+     *                  value. This is rather dangerous, as it might lead to the generation of duplicate ids. If
+     *                  set to <tt>false</tt>, the given <tt>nextValue</tt> has to be higher than the current sequence
+     *                  value.
+     */
+    public void setCounterValue(String sequence, long nextValue, boolean force) {
+        try {
+            // Select the current value which will be returned if all goes well....
+            if (oma.select(SequenceCounter.class).eq(SequenceCounter.NAME, sequence).exists()) {
+                if (force) {
+                    int updatedRows = oma.getDatabase()
+                                         .createQuery("UPDATE sequencecounter "
+                                                      + "SET nextValue = ${value} "
+                                                      + "WHERE name = ${name}")
+                                         .set("name", sequence)
+                                         .set("value", nextValue)
+                                         .executeUpdate();
+                    if (updatedRows == 1) {
+                        return;
+                    }
+                } else {
+                    int updatedRows = oma.getDatabase()
+                                         .createQuery("UPDATE sequencecounter "
+                                                      + "SET nextValue = ${value} "
+                                                      + "WHERE name = ${name} "
+                                                      + "  AND nextValue <= ${value}")
```

Comment:
couldnt just this line bei added to the query string if force is false so the whole if else is a lot sleeker?